services/brig/test/integration/API/Provider.hs

```diff
@@ -358,27 +358,73 @@ testListServicesByTagAndPrefix config db brig = do
+        return ls
+    getPage uid Nothing (Just start) size = do
+        rs <- listServiceProfilesByPrefix brig uid start size <!! const 200 === statusCode
         let Just ls = serviceProfilePageResults <$> decodeBody rs
         return ls
+    getPage _ Nothing Nothing _ = error "Query not supported"
 
-    searchAndAssert uid tags qry size expects = do
+    searchAndAssertNameChange :: ProviderId -> UserId -> Maybe MatchAny -> Text -> Service -> Http ()
+    searchAndAssertNameChange pid uid tags uniq svc = do
+        let sid = serviceId svc
+        Just svp <- decodeBody <$> (getServiceProfile brig uid pid sid
+                               <!!  const 200 === statusCode)
+        let origName = serviceProfileName svp
+        searchAndAssertWithSid uid tags origName 10 [origName] [sid]
+
+        let newName = mkName uniq "Wire"
+        let _upd = emptyUpdateService { updateServiceName = Just newName }
+        updateService brig pid (serviceId svc) _upd !!! const 200 === statusCode
+
+        -- Now we should find no such service with the original name, only with the new name
+        searchAndAssertWithSid uid tags origName 10 []        []
+        searchAndAssertWithSid uid tags newName  10 [newName] [sid]
+
+        -- Let's rollback
+        let _upd = emptyUpdateService { updateServiceName = Just origName }
+        updateService brig pid (serviceId svc) _upd !!! const 200 === statusCode
+
+        -- Searching the new name should return nothing
+        searchAndAssertWithSid uid tags newName  10 []         []
+        searchAndAssertWithSid uid tags origName 10 [origName] [sid]
+
+    searchAndAssertWithSid uid tags qry size expectsNames expectsSids = do
+        -- Test _with_ and _without_ tags and check we got the right sid back
+        _ls <- searchAndAssert uid tags qry size expectsNames
+        liftIO $ assertEqual ("check sid size: " ++ show qry) (length expectsSids) (length _ls)
+        let _sids = map serviceProfileId _ls
+        liftIO $ assertEqual ("check sid str: " ++ show qry) expectsSids _sids
+
+    searchAndAssert uid tags qry size expectsNames = do
+        -- Search with tag and name
         _ls <- getPage uid tags (Just qry) size
-        liftIO $ assertEqual ("size: " ++ show qry) (length expects) (length _ls)
+        liftIO $ assertEqual ("get page with tag, size: " ++ show qry) (length expectsNames) (length _ls)
+        let _names = map serviceProfileName _ls
+        liftIO $ assertEqual ("get page with tag, str: " ++ show qry) expectsNames _names
+        -- Search with name only
+        _ls <- getPage uid Nothing (Just qry) size
+        liftIO $ assertEqual ("get page no tag, size: " ++ show qry) (length expectsNames) (length _ls)
         let _names = map serviceProfileName _ls
-        liftIO $ assertEqual ("str: " ++ show qry) expects _names
+        liftIO $ assertEqual ("get page no tag: " ++ show qry) expectsNames _names
+        return _ls
 
```

Comment:
How about a testcase searching with an empty prefix?

Reply:
I didn't add it because searching with no tags and no prefix (at least that was the intention of the API), we should return a 400.
Alternatively, we can do what I was aiming for at first which was: if no prefix and no tag, return random services, potentially from a whitelisted provider. Any thoughts?

Reply:
I think we should deny it on the API level, i.e. return 400. My only concern is, I don't think that's what the code currently does, that's why I was thinking a test could prove me wrong.

Reply:
That was fixed with the [previous commit](https://github.com/wireapp/wire-server/pull/197/commits/7885d789cda94d9dcace1de9b02f5b7fd833078b) and I just [added a test for this case](https://github.com/wireapp/wire-server/pull/197/commits/8e4e84f4146631e9d8a39cb69033b46847c74516)